src/Go/effects/netscriptGoImplementation.ts

```diff
@@ -179,8 +179,19 @@ export async function getAIMove(boardState: BoardState): Promise<Play> {
     await sleep(400);
     const aiUpdatedBoard = makeMove(boardState, result.x, result.y, GoColor.white);
 
-    // Handle the AI breaking. This shouldn't ever happen.
     if (!aiUpdatedBoard) {
+      // If the previous move was made by the AI, log it, even if it didn't come from this thread.
+      // This can happen if multiple scripts are run, or if the user opens or interacts with the gameboard UI while a script is running
```

Comment:
Maybe I'm being slow because it's the morning, but I don't fully understand what "this" is in this context, or why two *player* moves would cause the *ai* move to break.

Reply:
There is code in place to make sure the AI takes its move when it is its turn, even if the game was saved and loaded where the AI was in mid-thought. This can be triggered in some cases where the player script is actively playing, but the gameboard AI beats it to the punch and makes a move first. (Since the gameboard AI is not part of the player script or netscript , it's like a separate process).

Solution: If the netscript AI tries to make a move but something beat it to it, return white's last move instead of an error.